internal/elasticsearch/ingest/processors.go

```diff
@@ -36,6 +36,20 @@ func (p Pipeline) Processors() (procs []Processor, err error) {
 	return procs, nil
 }
 
+// Processors return the original list of processors in an ingest pipeline.
+func (p Pipeline) ProcessorsWithoutReroute() (procs []Processor, err error) {
```

Comment:
Nit. In case we do more manipulations in the future apart of adding reroute processors. 
```suggestion
func (p Pipeline) OriginalProcessors() (procs []Processor, err error) {
```